src/components/DataTable/index.jsx

```diff
@@ -0,0 +1,120 @@
+import { Component } from 'react';
+import { hot } from 'react-hot-loader';
+import { withStyles } from '@material-ui/core/styles';
+import Table from '@material-ui/core/Table';
+import TableBody from '@material-ui/core/TableBody';
+import TableCell from '@material-ui/core/TableCell';
+import TableHead from '@material-ui/core/TableHead';
+import TableRow from '@material-ui/core/TableRow';
+import TableSortLabel from '@material-ui/core/TableSortLabel';
+import { camelCase } from 'change-case';
+import { memoizeWith } from 'ramda';
+import { arrayOf, func, string, object } from 'prop-types';
+import sort from '../../utils/sort';
+
+@hot(module)
+@withStyles(theme => ({
+  table: {
+    marginTop: theme.spacing.unit * 3,
+    width: '100%',
+    overflowX: 'auto',
+  },
+}))
+class DataTable extends Component {
+  static propTypes = {
+    /**
+     * A function to execute for each row to render in the table.
+     * Will be passed a datum from the table data. The function
+     * should return the JSX necessary to render the given row.
+     */
+    renderRow: func,
+    /**
+     * A list of header names to use on the table starting from the left.
+     */
+    headers: arrayOf(string),
+    /**
+     * A list of objects to display. Each element in the list is represented
+     * by a row and each element's key-value pair represents a column.
+     */
+    items: arrayOf(object).isRequired,
+  };
+
+  state = {
+    sortBy: null,
+    sortDirection: null,
+  };
+
+  getTableData = memoizeWith(
```

Comment:
The memoization and sorting need to happen in the project view. That's because the memoization function can vary across tables.